plugins/module_utils/sdkmanager.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Stanislav Shamilov <shamilovstas@protonmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import re
+
+from ansible_collections.community.general.plugins.module_utils import cmd_runner_fmt
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner
+
+__state_map = {
+    "present": "--install",
+    "absent": "--uninstall"
+}
+
+# sdkmanager --help 2>&1 | grep -A 2 -- --channel
+__channel_map = {
+    "stable": 0,
+    "beta": 1,
+    "dev": 2,
+    "canary": 3
+}
+
+
+def __map_channel(channel_name):
+    if channel_name not in __channel_map:
+        raise ValueError("Unknown channel name '%s'" % channel_name)
+    return __channel_map[channel_name]
+
+
+def sdkmanager_runner(module, **kwargs):
+    return CmdRunner(
+        module,
+        command='sdkmanager',
+        arg_formats=dict(
+            state=cmd_runner_fmt.as_map(__state_map),
+            name=cmd_runner_fmt.as_list(),
+            installed=cmd_runner_fmt.as_fixed("--list_installed"),
+            list=cmd_runner_fmt.as_fixed('--list'),
+            newer=cmd_runner_fmt.as_fixed("--newer"),
+            sdk_root=cmd_runner_fmt.as_opt_eq_val("--sdk_root", ignore_none=True),
+            channel=cmd_runner_fmt.as_func(lambda x: ["{0}={1}".format("--channel", __map_channel(x))])
+        ),
+        force_lang="C.UTF-8",  # Without this, sdkmanager binary crashes
```

Comment:
Some platforms would moan about his - I want to say FreeBSD but not sure anymore. That happened to me when writing the `ansible_galaxy_install` (or maybe `snap`) module so I had to use `en_US.UTF-8`. There is a builtin function in Ansible that tries to search for the best locale to use, if you set this parameter here to `auto` it will use that, you might want to give it a try. In the future, that behaviour will be the default.

Reply:
As you can see from the comment, I added this because the `sdkmanager` tool crashes if the locale is set to `C`. What is more interesting is that while testing on VMs using vagrant, it was working fine until I started running tests on docker containers with `ansible-test integration`, and the tool was crashing on any of them. I still have no idea why exactly `C` locale was causing the tool to crash inside containers, as googling for the stacktrace generated by the crash didn't yield anything meaningful. Is there a chance that selecting `auto` would choose `C` locale and subsequently, cause the crash?

Reply:
A chance? yes, but not after it tried four other locales supporting UTF-8. See:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/locale.py#L34

I also find it strange that it fails in containers but not in the VMs, but I would be very surprised if the cause is related to that difference.

Reply:
I did some more testing and it seems I was wrong that it crashes on all test containers. To test this crash, no python code is required, only a playbook like the following one:
````yaml
---

- hosts: all
  vars:
    android_cmdline_temp_dir: "/tmp/cmdlinetools"
    android_sdk_location: "/tmp/androidsdk"
    commandline_tools_link: https://dl.google.com/android/repository/commandlinetools-linux-11076708_latest.zip
  tasks:
    - name: Install dependencies
      become: true
      apt:
        update_cache: true
        name:
          - openjdk-17-jre-headless
          - unzip
        state: present

    - name: Create Android SDK directory
      file:
        path: "{{ android_sdk_location }}"
        state: directory

    - name: Check that sdkmanager is installed
      stat:
        path: "{{ android_sdk_location }}/cmdline-tools/latest/bin/sdkmanager"
      register: sdkmanager_installed

    - name: Install Android command line tools
      when: not sdkmanager_installed.stat.exists
      block:
        - name: Create Android SDK dir structure
          file:
            path: "{{ item.path }}"
            state: "{{ item.state }}"
          with_items:
            - { path: "{{ android_cmdline_temp_dir }}", state: "directory" }
            - { path: "{{ android_sdk_location }}/cmdline-tools/latest", state: "directory" }

        - name: Download Android command line tools
          unarchive:
            src: "{{ commandline_tools_link }}"
            dest: "{{ android_cmdline_temp_dir }}"
            remote_src: yes
            creates: "{{ android_cmdline_temp_dir }}/cmdline-tools"
          when: not sdkmanager_installed.stat.exists


        - name: Fix directory structure
          copy:
            src: "{{ android_cmdline_temp_dir }}/cmdline-tools/"
            dest: "{{ android_sdk_location }}/cmdline-tools/latest"
            remote_src: yes


    - name: Try runnung sdkmanager
      environment:
        LC_ALL: "C"
        LANGUAGE: "C"
        PATH: "{{ ansible_env.PATH }}:{{ android_sdk_location }}/cmdline-tools/latest/bin"
      shell: "sdkmanager --list"
````
The crash happens in the "Try running sdkmanager" task when `LC_ALL` and `LANGUAGE` environment variable are set to `C`.
If I run this playbook on a VM (`ubuntu/focal64` Vagrant box), the playbook runs fine. However, if it is run as a test for `ansible-test integration %TARGET% --docker`, `sdkmanager` tool will crash ([crash log](https://github.com/user-attachments/files/18111897/log.txt) attached). %TARGET% is the name of the directory where you'll put the playbook to be run as a test (just like normal tests are run).
What's interesting, despite the fact that the default test container reports that it is Ubuntu 20.04, running `ansible-test integration %TARGET% -v --docker ubuntu2004` does not crash.

Reply:
The containers used in integration tests have some tweaking done to them. Possibly/likely some of it is different in the VM. Anyways, have you by any chance tried testing with `auto`?

Reply:
Yes, seems to be working ok. Pushing the change now

Reply:
@russoz Apparently, it does not work ok. Some of the CI jobs failed, due to `sdkmanager` crashing